### ISSUE TITLE
Error: conditional_effects() and conditional_smooths() stopped working

### ISSUE THREAD

**Issue by ZoeGold** [paul-buerkner/brms#1515]
Several months ago I wrote a script running several GAMs in `brms`. Now when running the script again (with the saved model RDS outputs, so nothing changed there) I am suddenly running into an error where conditional_effects() and conditional_smooths() return 

```
Error in h(simpleError(msg, call)) : 
  error in evaluating the argument 'x' in selecting a method for function 't': the dims contain missing values`
```

Additionally, posterior_smooths() also sometimes returns the same error, but this seems to be more variable (it occurs with brms v. 2.19.6 but not 2.19.0 or 2.17.0)

There are no missing values in the data or model predictions, this code with these exact models did work before, and the problem appears to be limited to my computer. Since last running this code, I have updated R and some packages (see session info below). My attempts to trouble shoot led me to identify the following:

1. I tried various versions of `rstan` (currently on the source version as this is recommended with R 4.3). rstan also works as it should and compiles properly.
2. I tried older versions of `brms` (2.17.0 & 2.19.0) but this also did not resolve the issue
3. This issue already existed when I was on R version 4.2 (I updated to 4.3 in the hope that this would resolve it)
4. My hunch is that it might have to do with some kind of mismatch between rstan and brms, or brms and the R version. 

Conditional_effects() and conditional_smooths() did work on another computer running older versions of R, rstan, and brms (see second session info below), but I haven't been able to pinpoint which of the three is the crucial factor here. I would be fine with rolling back to R v 4.1 or even older, but wouldn't want to do this if this is not likely the root of the issue. 

I hope I provided enough information here, and thank you for your help! 

**Session info my PC (where it does not work)**
```
R version 4.3.1 (2023-06-16 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_Germany.utf8  LC_CTYPE=English_Germany.utf8    LC_MONETARY=English_Germany.utf8 LC_NUMERIC=C                     LC_TIME=English_Germany.utf8    

time zone: Europe/Berlin
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] dplyr_1.1.2         cowplot_1.1.1       ggnewscale_0.4.9    ggplot2_3.4.2       reshape2_1.4.4      akima_0.6-3.4       matrixStats_1.0.0   brms_2.19.6         Rcpp_1.0.10         fitdistrplus_1.1-11 survival_3.5-5      MASS_7.3-60        
[13] viridis_0.6.3       viridisLite_0.4.2  

loaded via a namespace (and not attached):
 [1] gridExtra_2.3        remotes_2.4.2        inline_0.3.19        rlang_1.1.1          magrittr_2.0.3       compiler_4.3.1       mgcv_1.8-42          loo_2.6.0            callr_3.7.3          vctrs_0.6.3          stringr_1.5.0        pkgconfig_2.0.3     
[13] crayon_1.5.2         fastmap_1.1.1        backports_1.4.1      ellipsis_0.3.2       utf8_1.2.3           threejs_0.3.3        cmdstanr_0.5.3       promises_1.2.0.1     markdown_1.7         ps_1.7.5             xfun_0.39            jsonlite_1.8.5      
[25] later_1.3.1          parallel_4.3.1       prettyunits_1.1.1    R6_2.5.1             dygraphs_1.1.1.6     stringi_1.7.12       StanHeaders_2.26.27  rstan_2.26.22        knitr_1.43           zoo_1.8-12           base64enc_0.1-3      bayesplot_1.10.0    
[37] httpuv_1.6.11        Matrix_1.5-4.1       splines_4.3.1        igraph_1.5.0         tidyselect_1.2.0     abind_1.4-5          codetools_0.2-19     miniUI_0.1.1.1       curl_5.0.1           processx_3.8.1       pkgbuild_1.4.1       lattice_0.21-8      
[49] tibble_3.2.1         plyr_1.8.8           shiny_1.7.4          withr_2.5.0          bridgesampling_1.1-2 posterior_1.4.1      coda_0.19-4          desc_1.4.2           RcppParallel_5.1.7   xts_0.13.1           pillar_1.9.0         tensorA_0.36.2      
[61] checkmate_2.2.0      DT_0.28              stats4_4.3.1         shinyjs_2.1.0        distributional_0.3.2 generics_0.1.3       rprojroot_2.0.3      sp_1.6-1             rstantools_2.3.1     munsell_0.5.0        scales_1.2.1         gtools_3.9.4        
[73] xtable_1.8-4         glue_1.6.2           tools_4.3.1          shinystan_2.6.0      colourpicker_1.2.0   mvtnorm_1.2-2        grid_4.3.1           crosstalk_1.2.0      colorspace_2.1-0     nlme_3.1-162         cli_3.6.1            fansi_1.0.4         
[85] Brobdingnag_1.2-9    V8_4.3.0             gtable_0.3.3         digest_0.6.31        htmlwidgets_1.6.2    farver_2.1.1         htmltools_0.5.5      lifecycle_1.0.3      mime_0.12            shinythemes_1.2.0
```

**Session info other PC (where conditional_effects does work)**
```
R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 11.6

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] dplyr_1.1.1         cowplot_1.1.1       ggnewscale_0.4.9    ggplot2_3.4.2       reshape2_1.4.4      akima_0.6-3.4       matrixStats_1.0.0   brms_2.17.0         Rcpp_1.0.7          fitdistrplus_1.1-11 survival_3.2-11    
[12] MASS_7.3-54         viridis_0.6.2       viridisLite_0.4.2   cmdstanr_0.5.3     

loaded via a namespace (and not attached):
 [1] nlme_3.1-152         xts_0.12.1           threejs_0.3.3        rstan_2.21.2         tensorA_0.36.2       tools_4.1.1          backports_1.4.1      utf8_1.2.3           R6_2.5.1             DT_0.19              mgcv_1.8-36         
[12] colorspace_2.1-0     sp_1.4-5             withr_2.5.0          tidyselect_1.2.0     gridExtra_2.3        prettyunits_1.1.1    processx_3.8.1       Brobdingnag_1.2-7    curl_4.3.2           compiler_4.1.1       cli_3.6.1           
[23] shinyjs_2.0.0        isoband_0.2.7        labeling_0.4.2       colourpicker_1.1.1   posterior_1.4.1      scales_1.2.1         dygraphs_1.1.1.6     checkmate_2.2.0      mvtnorm_1.1-3        ggridges_0.5.3       callr_3.7.0         
[34] StanHeaders_2.21.0-7 stringr_1.4.0        digest_0.6.31        base64enc_0.1-3      pkgconfig_2.0.3      htmltools_0.5.2      fastmap_1.1.0        htmlwidgets_1.5.4    rlang_1.1.1          rstudioapi_0.13      shiny_1.7.1         
[45] farver_2.1.1         generics_0.1.3       zoo_1.8-9            jsonlite_1.8.5       crosstalk_1.1.1      gtools_3.9.2         distributional_0.3.2 inline_0.3.19        magrittr_2.0.3       loo_2.4.1            bayesplot_1.8.1     
[56] Matrix_1.3-4         munsell_0.5.0        fansi_1.0.4          abind_1.4-5          lifecycle_1.0.3      stringi_1.7.5        pkgbuild_1.2.0       plyr_1.8.6           grid_4.1.1           parallel_4.1.1       promises_1.2.0.1    
[67] crayon_1.4.1         miniUI_0.1.1.1       lattice_0.20-44      splines_4.1.1        knitr_1.36           ps_1.7.5             pillar_1.9.0         igraph_1.2.6         markdown_1.1         shinystan_2.5.0      codetools_0.2-18    
[78] stats4_4.1.1         rstantools_2.1.1     glue_1.6.2           V8_3.4.2             data.table_1.14.8    RcppParallel_5.1.4   vctrs_0.6.3          httpuv_1.6.3         gtable_0.3.3         xfun_0.26            mime_0.12           
[89] xtable_1.8-4         coda_0.19-4          later_1.3.0          rsconnect_0.8.24     tibble_3.2.1         shinythemes_1.2.0    ellipsis_0.3.2       bridgesampling_1.1-2
```

**Comment by ZoeGold** [comment]
Yes, sorry it took a while:

```
data = data.frame(
  n = rpois(100, 5),
  distcoast_z = runif(100, -1, 5),
  tidedif_z = rnorm(100, 0, 2),
  locationfactor = factor(rep(c("A","B", "C", "D", "E"),each=20)),
  toolusers = factor(rep(c("yes","no"), each = 50))
)

prior <- c(prior(normal(0, 2), class = Intercept),
                 prior(normal(0,2), class = b),
                 prior(exponential(1), class = sds))

fit1 <- brm(n  ~ t2(tidedif_z, distcoast_z, bs = c("cc", "tp"), k = c(10, 6), full = TRUE) +
                   t2(tidedif_z, distcoast_z, bs = c("cc", "tp"), by = toolusers, k = c(10, 6), m = 1) + toolusers +
                   s(locationfactor, bs = "re"), family = poisson(),  knots = list(tidedif_z =c(-1.8,1.8)),  data = data, 
                 chain = 2, core = 2, iter = 2000, control = list(adapt_delta = 0.99, max_treedepth = 12), backend = "cmdstanr", prior = prior)


summary(fit1)
plot(conditional_smooths(fit1))
```

**Comment by paul-buerkner** [comment]
Yes, I can confirm this error. Will work on a fix.

**Comment by paul-buerkner** [comment]
The problem is the `t2(tidedif_z, distcoast_z, bs = c("cc", "tp"), by = toolusers, k = c(10, 6), m = 1)` spline I think. If I set m > 0, thinks work out well. Does the same work with gamm or gamm4 packages?

**Comment by ZoeGold** [comment]
Thank you for getting to this so quickly! I haven't worked with the gamm or gamm4 package before, so I am not sure if this construction would work there. It's interesting that this formula worked previously with conditional_effects but now stopped working.  I'll look into the difference between m =1 and m > 0 and see if I can otherwise switch to m > 0

**Comment by paul-buerkner** [comment]
The syntax should be the same as in brms, so it should not be too cumbersome to test. I am not sure why it started failing recently with new brms version, since the error happens within mgcv. I will try to look at this by comparing brms 2.19.0 with 2.19.6.

**Comment by ZoeGold** [comment]
Thank you! I tried running the same model with the same syntax in both mgcv and gamm4, both seems to work without issue (see below). I hope you find out what causes the bug in conditional_effects() and conditional_smooths() by comparing brms 2.19.0 with 2.19.6. In the meantime, I will work in an older version of R and brms to avoid the issue. Thank you again for your time! 

```
data = data.frame(
  n = rpois(1000, 5),
  distcoast_z = runif(1000, -1, 5),
  tidedif_z = rnorm(1000, 0, 2),
  locationfactor = factor(rep(c("A","B", "C", "D", "E"),each=200)),
  toolusers = factor(rep(c("yes","no"), each = 500))
)

prior <- c(prior(normal(0, 2), class = Intercept),
           prior(normal(0,2), class = b),
           prior(exponential(1), class = sds))

fit1 <- brm(n  ~ t2(tidedif_z, distcoast_z, bs = c("cc", "tp"), k = c(10, 6), full = TRUE) +
              t2(tidedif_z, distcoast_z, bs = c("cc", "tp"), by = toolusers, k = c(10, 6), m = 1) + toolusers +
              s(locationfactor, bs = "re"), family = poisson(),  knots = list(tidedif_z =c(-1.8,1.8)),  data = data, 
            chain = 2, core = 2, iter = 2000, control = list(adapt_delta = 0.99, max_treedepth = 12), backend = "cmdstanr", prior = prior)


summary(fit1)
plot(conditional_smooths(fit1))

require(mgcv)
fit1_mgcv <- gam(n  ~ t2(tidedif_z, distcoast_z, bs = c("cc", "tp"), k = c(10, 6), full = TRUE) +
                   t2(tidedif_z, distcoast_z, bs = c("cc", "tp"), by = toolusers, k = c(10, 6), m = 1) + toolusers +
                   s(locationfactor, bs = "re"), family = poisson(),  knots = list(tidedif_z =c(-1.8,1.8)),  data = data, method = "REML")
summary(fit1_mgcv)
plot(fit1_mgcv)

require(gamm4)
fit1_gamm4 <- gamm4(n  ~ t2(tidedif_z, distcoast_z, bs = c("cc", "tp"), k = c(10, 6), full = TRUE) +
                   t2(tidedif_z, distcoast_z, bs = c("cc", "tp"), by = toolusers, k = c(10, 6), m = 1) + toolusers +
                   s(locationfactor, bs = "re"), family = poisson(),  knots = list(tidedif_z =c(-1.8,1.8)),  data = data)
summary(fit1_gamm4$gam)
summary(fit1_gamm4$mer)
plot(fit1_gamm4$gam)
```

**Comment by paul-buerkner** [comment]
The same error happens for me in brms 2.19.0, leaving the rest of the packages versions the same. Did you really have everything else the same in both setups? 

While I have made some changes to splines between 2.19.0 and 2.19.6, I don't think I have changed anythig that could explain this behavior. My feeling is that this may be a bug in mgcv but one that only comes up when interfaced with brms. I might be wrong of course.

**Comment by ZoeGold** [comment]
I had the same error in brms 2.19.0 as well, the only difference between 2.19.0 and 2.19.6 is that posterior_predict() does (sometimes) work in 2.19.0 and never in 2.19.6. But conditional_effects() and conditional_smooths() both throw the same error in version 2.19.0 and 2.19.6.

My colleagues' computer indeed has a lot of other packages, it's not just brms that is older (v. 2.17.0), which is why I also can't narrow down where the problem comes from exactly.

**Comment by paul-buerkner** [comment]
What are the mgcv versions on the different machines? I think this is the likely cause of the different behavior.

**Comment by ZoeGold** [comment]
The machine with the error is running mgcv 1.8.42, the machine without the error is on 1.8.36.

**Comment by paul-buerkner** [comment]
Could that be the reason then? What if you install 1.8.36 on the machine having the error?

**Comment by paul-buerkner** [comment]
I have informed Simon Wood about this issue and hope we can fix it from the mgcv side. Since this is related to mgcv internals, there is not really anything I can do to fix it from the brms side at the moment.

**Comment by paul-buerkner** [comment]
fixed in mgcv version 1.9.0

[mgcv_1.9-0.tar.gz](https://github.com/paul-buerkner/brms/files/11881444/mgcv_1.9-0.tar.gz)
